openlibrary/solr/update_work.py

```diff
@@ -327,7 +327,7 @@ def process_editions(self, w, editions, ia_metadata, identifiers):
                         v = v.strip()
                         if v not in identifiers[k]:
                             identifiers[k].append(v)
-        return sorted(editions, key=lambda e: e.get('pub_year', None))
+        return sorted(editions, key=lambda e: int(e.get('pub_year', 0)) or -sys.maxsize)
```

Comment:
This does have the slight caveat that it won't work if the year actually is 0 😅 I think it's ok to just pass the max through int:

```suggestion
        return sorted(editions, key=lambda e: int(e.get('pub_year') or -sys.maxsize))
```